### ISSUE TITLE
QA Loss Cleanup

### ISSUE THREAD

**Issue by sshleifer** [huggingface/transformers#6204]
This snippet appears a lot of places and could be factored out into a `calc_qa_loss(logits)`

Requires some care, because I'm not sure how good the test coverage is, and if it doesn't improve readability we shouldn't do it.

```python
        logits = self.qa_outputs(sequence_output)
        start_logits, end_logits = logits.split(1, dim=-1)
        start_logits = start_logits.squeeze(-1)
        end_logits = end_logits.squeeze(-1)

        outputs = (start_logits, end_logits,) + outputs[2:]
        if start_positions is not None and end_positions is not None:
            # If we are on multi-GPU, split add a dimension
            if len(start_positions.size()) > 1:
                start_positions = start_positions.squeeze(-1)
            if len(end_positions.size()) > 1:
                end_positions = end_positions.squeeze(-1)
            # sometimes the start/end positions are outside our model inputs, we ignore these terms
            ignored_index = start_logits.size(1)
            start_positions.clamp_(0, ignored_index)
            end_positions.clamp_(0, ignored_index)

            loss_fct = CrossEntropyLoss(ignore_index=ignored_index)
            start_loss = loss_fct(start_logits, start_positions)
            end_loss = loss_fct(end_logits, end_positions)
            total_loss = (start_loss + end_loss) / 2
```

@stas00

**Comment by stas00** [comment]
I will experiment. Thank you!

**Comment by stas00** [comment]
See: https://github.com/huggingface/transformers/pull/6430

**Comment by stas00** [comment]
So I guess we can close this one, since the experiment wasn't accepted.
